File: 2321-maximum-score-of-spliced-array/2321-maximum-score-of-spliced-array.py
class Solution:
    def maximumsSplicedArray(self, nums1: List[int], nums2: List[int]) -> int:
        n = len(nums1)
        sumOfNums1 = sum(nums1)
        sumOfNums2 = sum(nums2)
        
        ans = max(sumOfNums1,sumOfNums2)
        
        tmp = [0]*n
        for i in range(0,n):
            tmp[i] = nums1[i] - nums2[i]
            
        worst = 0
        total = 0
        for i in range(0,n):
            if total + tmp[i] > 0:
                total = 0
            else:
                total += tmp[i]
                
            worst = min(worst,total)
        
        ans = max(ans,sumOfNums1  - worst)
        
        for i in range(0,n):
            tmp[i] = -tmp[i]
            
        worst = 0
        total = 0
        for i in range(0,n):
            if total + tmp[i] > 0:
                total = 0
            else:
                total += tmp[i]
                
            worst = min(worst,total)
        
        ans = max(ans,sumOfNums2  - worst)
        
        return ans

File: 2321-maximum-score-of-spliced-array/2321-maximum-score-of-spliced-array.cpp
class Solution {
public:
    
     int funct(vector<int> &nums1, vector<int> &nums2, int n) {
         vector<int> tmp(n);
         int total = 0;
         for(int i = 0; i < n; i++) {
             tmp[i] = nums1[i] - nums2[i];
             total += nums1[i];
         }
         int worst = 0, curr = 0;
         for(int i = 0; i < n; i++) {
             curr = min(tmp[i],curr + tmp[i]);
             worst = min(worst,curr);
         }
         if(worst > 0) {
             return total;
         }
         return total - worst;
     }
    
    int maximumsSplicedArray(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size();
        return max(funct(nums1,nums2,n),funct(nums2,nums1,n));
    }
    
};